translib/transformer/xfmr_intf.go

```diff
@@ -3144,3 +3212,326 @@ func retrievePortChannelAssociatedWithIntf(inParams *XfmrParams, ifName *string)
 	}
 	return nil, err
 }
+
+var YangToDb_intf_type_xfmr FieldXfmrYangToDb = func(inParams XfmrParams) (map[string]string, error) {
+	res_map := make(map[string]string)
+
+	pathInfo := NewPathInfo(inParams.uri)
+	ifName := pathInfo.Var("name")
+	intfType, _, err := getIntfTypeByName(ifName)
+	if intfType == IntfTypeUnset || err != nil {
+		log.Info("DbToYang_intf_type_xfmr - Invalid interface type IntfTypeUnset")
+		return res_map, errors.New("Invalid interface type IntfTypeUnset")
```

Comment:
1) Use specific error message
2) Avoid exposing internal enum name `IntfTypeUnset` in the error message